pkg/local_object_storage/metabase/inhume.go

```diff
@@ -42,6 +45,25 @@ func (i InhumeRes) DeletedLockObjects() []oid.Address {
 	return i.deletedLockObj
 }
 
+// GetDeletionInfoLength returns amount of stored elements
+// in deleted sizes array.
+func (i InhumeRes) GetDeletionInfoLength() int {
+	return len(i.sizes)
+}
+
+// GetDeletionInfoByIndex returns both deleted object sizes and
+// associated container ID by index.
+func (i InhumeRes) GetDeletionInfoByIndex(target int) (cid.ID, uint64) {
+	return i.containerIds[target], i.sizes[target]
+}
+
+// StoreDeletionInfo stores size of deleted object and associated container ID
+// in corresponding arrays.
+func (i *InhumeRes) StoreDeletionInfo(containerID cid.ID, deletedSize uint64) {
+	i.sizes = append(i.sizes, deletedSize)
+	i.containerIds = append(i.containerIds, containerID)
```

Comment:
We use and append them together, why not store them together? (define a private struct, return 2 values as you already do).